src/converter.js

```diff
@@ -178,32 +179,38 @@ converter._object = (name, node, mode) => {
         node
       )
     }
-    if (!_.isPlainObject(node.properties)) {
-      throw new SchemaError('No properties defined for object', node)
+    if (!hasProperties) {
+      // Big Query can handle semi-structured data :
+      // https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json#loading_semi-structured_json_data
+      fieldType = 'JSON'
     }
-    if (Object.keys(node.properties).length === 0) {
+
+    if (hasProperties && Object.keys(node.properties).length === 0) {
       throw new SchemaError(
         'Record fields must have one or more child fields',
         node
       )
     }
-    const required_properties = node.required || []
-    const properties = node.properties
 
-    let fields = Object.keys(properties).map((key) => {
-      const required = required_properties.includes(key)
-        ? 'REQUIRED'
-        : 'NULLABLE'
-      return converter._visit(key, properties[key], required)
-    })
+    result = converter._scalar(name, fieldType, mode, node.description)
```

Comment:
We could move this up to inside the if at line 182 and return early, this would avoid another `if (hasProperties)` check below